src/Architectures.jl

```diff
@@ -61,7 +61,11 @@ end
 
 Run Oceananigans on Reactant.
 """
-struct ReactantState <: AbstractSerialArchitecture end
+struct ReactantState{S} <: AbstractSerialArchitecture
+    sharding::S
+end
+
+ReactantState() = ReactantState(missing)
```

Comment:
should we allow this?

trivial question: `missing` vs `nothing`? The difference in mind is: is `sharding` _required_? If sharding is required then `missing` conveys the right message. If we can also run with "no sharding" then I think `nothing` is clearer, and also consistent with what we typically use in Oceananigans for "not" doing something.

Reply:
By default it should be `NoSharding` (or `nothing`) so I think it makes sense to default to nothing. I will update it